lib/workers/repository/process/lookup/index.ts

```diff
@@ -224,6 +225,14 @@ export async function lookupUpdates(
           newMajor: versioning.getMajor(currentVersion)!,
         });
       }
+
+      if (
+        rangeStrategy !== 'update-lockfile' &&
+        isNpmXRangeAll(currentValue, config)
```

Comment:
I prefer to keep manager and versioning-specific logic out of lib/workers.

I think we already support this for bundler and composer, so I'm trying to locate the logic they use.

When supporting this for Composer it was very small: https://github.com/renovatebot/renovate/pull/12930

I hd a feeling this code is related: https://github.com/renovatebot/renovate/blob/c22380d65cd25bf5d11f8e8c8c50bbd990f6de07/lib/workers/repository/process/lookup/index.ts#L203-L205

Reply:
> What goes wrong if you don't add the isNpmXRangeAll logic?

We will get additional ["updates"](https://github.com/renovatebot/renovate/issues/17916#issuecomment-1273654468) for strategy pin and bump.


`currentVersion` is determined by `getCurrentVersion()` 
https://github.com/renovatebot/renovate/blob/c22380d65cd25bf5d11f8e8c8c50bbd990f6de07/lib/workers/repository/process/lookup/index.ts#L185-L204

In `getCurrentVersion()` we have
https://github.com/renovatebot/renovate/blob/e49f06ea15be68195e3f0743d18d67782470fbf6/lib/workers/repository/process/lookup/current.ts#L33-L45

#### Summary per strategy:

`pin`: `currentVersion=lockedVersion|latest`, so in case we are using the locked version we will get extra updates based on this value. 
`bump`: `currentVersion` will be set to the first release of a given dependency as the is range=`'*'`. so we will get all updates in between, e.g. one very outdated update to the next minor and probably another one to latest.
`replace, widen`: `currentVersion=latest`, no updates in this case.

Above is the reason for the added logic for skipping updates for non `lockfile-update` strategies.

I not entirely sure about the composer flow, but in our case `!currentVersion!` will be `false` so we wont return there.
and thats a good thing i guess, because otherwise we will skip the `isPin` update otherwise.

Reply:
If we dont skip these at the lookup phase, is there a better place for skipping these and not apply them?

Reply:
If rangeStrategy=pin and currentValue=* then I'd expect only a single pin PR regardless of lock file presence or not. What "extra" do you get?

If rangeStrategy=bump, and currentValue=* then I'm not sure what that combination should even do. I think it can be ok to even ignore it and leave as-is.

Reply:
> If rangeStrategy=pin and currentValue=* then I'd expect only a single pin PR regardless of lock file presence or not. What "extra" do you get?

From https://github.com/renovatebot/renovate/issues/17916#issuecomment-1268919101: 
### Repro with lock file and pin replace strategy -
All dependencies are pinned to their lockfile's value.
 - https://github.com/ladzaretti/renovate-star-dependencies/pull/1

> What "extra" do you get?

### These are the "extra" PRs
These are basically just like `update-lockfile" but the dependencies are also getting pinned.

dont mind the `autoclosed`, these are closed because of the `isNpmXRangeAll` login in the `lookup/index.ts`. 

 - https://github.com/ladzaretti/renovate-star-dependencies/pull/4
 - https://github.com/ladzaretti/renovate-star-dependencies/pull/3

tbh, above make sense to me ¯\\_(ツ)_/¯

Reply:
as for the bump, I suggest bumping the range from `'*'` (:=`>=0.0.0`) to `'>=newVersion' 

as per https://github.com/renovatebot/renovate/issues/17916#issuecomment-1273160388

> e.g.
let rangeStrategy="bump", newVersion = "1.0.0";
-> upgrade offering : "*" -> ">=1.0.0"

if we do offer the extra PRs for the pinning strat, and bumping the range as in this comment than we can remove the `isNpmRangeAll` from `lookup/index.ts` as desired.

Reply:
doesn't renovate skip the update when `getNewValue` from versioning returns null? so we can return null for bump and * ?